BlueprintUI/Sources/Layout/Alignment.swift

```diff
@@ -0,0 +1,106 @@
+import CoreGraphics
+
+/// Types used to identify alignment guides.
+///
+/// Types conforming to `AlignmentID` have a corresponding alignment guide value,
+/// typically declared as a static constant property of `HorizontalAlignment` or 
+/// `VerticalAlignment`.
+///
+public protocol AlignmentID {
+
+    /// Returns the value of the corresponding guide, in `context`, when not
+    /// otherwise set in `context`.
+    static func defaultValue(in context: ElementDimensions) -> CGFloat
+}
+
+/// An alignment position along the horizontal axis.
+public struct HorizontalAlignment: Equatable {
+
+    var id: AlignmentID.Type
+
+    /// Creates an instance with the given ID.
+    ///
+    /// Note: each instance should have a unique ID.
+    public init(_ id: AlignmentID.Type) {
+        self.id = id
+    }
+
+    public static func == (lhs: HorizontalAlignment, rhs: HorizontalAlignment) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
+
+/// An alignment position along the vertical axis.
+public struct VerticalAlignment: Equatable {
+
+    var id: AlignmentID.Type
+
+    /// Creates an instance with the given ID.
+    ///
+    /// Note: each instance should have a unique ID.
+    public init(_ id: AlignmentID.Type) {
+        self.id = id
+    }
+
+    public static func == (lhs: VerticalAlignment, rhs: VerticalAlignment) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
+
+extension HorizontalAlignment {
+    enum Leading: AlignmentID {
+        static func defaultValue(in context: ElementDimensions) -> CGFloat {
+            0
+        }
+    }
+
+    /// A guide marking the leading edge of the element.
+    public static let leading = HorizontalAlignment(Leading.self)
+
+    enum Center: AlignmentID {
+        static func defaultValue(in d: ElementDimensions) -> CGFloat {
```

Comment:
Thought: Is there a more descriptive name for this than `value`? Seeing `d.width / 2` here, I'm not sure what value the halved width is getting applied to. The centerpoint, the min axis, the max axis, etc.

Reply:
(I guess it's really dependent on the context of the alignment, so maybe no better name?)

Reply:
Copying the SwiftUI API here. I'm not sure what would be a better name though, since as you say, this is just an alignment value, and its use depends on the context.